recipe/meta.yaml

```diff
@@ -33,22 +33,28 @@ requirements:
     - python
     - cython >=3.0.0,<3.1.0
     - extension-helpers >=1,<2
-    - numpy 1.23 # [py==310]
-    - numpy {{ numpy }} # [py>=311]
+    - numpy >=2.0.0
     - pip
     - setuptools
     - setuptools_scm
     - wheel
   run:
     - python
-    - {{ pin_compatible("numpy") }}
+    - numpy >=1.23
     - pyerfa >=2.0.1.1
-    - astropy-iers-data >=0.2024.4.29.0.28.48
+    - astropy-iers-data >=0.2024.7.29.0.32.7
     - pyyaml >=3.13
     - packaging >=19.0
-  run_constrained:
     - matplotlib-base >=3.3,!=3.4.0,!=3.5.2
     - scipy >=1.8
+  run_constrained:
+    - typing_extensions >=4.0.0
+    - pyarrow >=7.0.0
+    - asdf-astropy >=0.3
+    - ipython >=4.2
+    - pytest >=7.0
+    - fsspec[http] >=2023.4.0
```

Comment:
It seems the only extra requirement for `fsspec[http]` is `requests`. Although there's no specific boundary, I've added it explicitly, but am also okay with removing it. It would be nice if conda did recognize that but I'm sure there would be a LOT of logic/hardcoding to cover all the different cases and to know which packages need to automatically be included based off that syntax.